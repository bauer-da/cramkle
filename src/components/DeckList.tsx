import * as React from 'react'
import { connect } from 'react-redux'

import Deck from './Deck'
import { DeckEntity } from '../types'
import { fetchDecksRequest } from '../actions/deck'
import { loadingDecks, getDecks } from '../selectors/deck'

interface DeckListProps {
  requestDecks: () => {}
  decks: [DeckEntity]
  loading: boolean
}

class DeckList extends React.Component<DeckListProps> {
  componentDidMount() {
    this.props.requestDecks()
  }

  render() {
    const { loading, decks } = this.props

    if (loading) {
      return <p>loading...</p>
    }

    if (decks.length as number === 0) {
      return (
        <p className="mdc-typography--caption">
          You haven&apos;t created any decks yet
        </p>
      )
    }

    return (
      <div className="mdc-layout-grid">
        <p className="mdc-typography--caption">Decks</p>

        <div className="mdc-layout-grid__inner">
          {decks.map(deck => (
            <div key={deck.deckId} className="mdc-layout-grid__cell">
              <Deck {...deck} />
            </div>
          ))}
        </div>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  decks: getDecks(state),
  loading: loadingDecks(state),
})

const mapDispatchToProps = dispatch => ({
  requestDecks: () => dispatch(fetchDecksRequest()),
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DeckList)
