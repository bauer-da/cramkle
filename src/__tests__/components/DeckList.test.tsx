import * as React from 'react'
import { render } from 'react-testing-library'
import { createStore } from 'redux'

import rootReducer from '../../rootReducer'
import DeckList from '../../components/DeckList'

describe('<DeckList />', () => {
  it('should match snapshot', () => {
    const dummyStore = createStore(rootReducer)

    const component = render(<DeckList store={dummyStore} />)

    expect(component.container).toMatchSnapshot()
  })
})
