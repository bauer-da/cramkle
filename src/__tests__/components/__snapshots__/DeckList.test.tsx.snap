// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeckList /> should match snapshot 1`] = `
<div>
  <p>
    loading...
  </p>
</div>
`;
